### PR TITLE
[TNA][Review] Review of the New Architecture guide

### PR DESCRIPTION
**Pull request by cipolleschi**
This Pr is a first review of the New Architecture guide.

The main focus is to review the Playbook having in mind the release of 0.70.0.

It also tries to rename all the `new architecture` to `New Architecture` and all the `CodeGen` and `code-gen` to `Codegen`.

## TODO

- [x] review Enabling TurboModule on Android with Autolinking
- [x] review Enabling Fabric Components on Android with Autolinking
- [x] add banners on Native Modules and Native Components

### CODE REVIEW

File: docs/new-architecture-app-intro.md [hunk 1]
```diff
@@ -7,40 +7,25 @@ import NewArchitectureWarning from './\_markdown-new-architecture-warning.mdx';
...
 
-As of this writing, the latest stable release is `0.67.2`. Once you have upgraded your project to this version successfully, you may proceed to targeting the `0.0.0-20220201-2008-79975d146` nightly release. You may target this nightly release the same way you’d target any other version of React Native:
+As of this writing, the latest stable release is `0.69.1`. Once you have upgraded your project to this version successfully, you may proceed to targeting the `0.0.0-20220705-2052-71e89213a` nightly release. You may target this nightly release the same way you’d target any other version of React Native:
 
 ```bash
-yarn add react-native@0.0.0-20220201-2008-79975d146
+yarn add react-native@0.0.0-20220705-2052-71e89213a
 ```
 
-## Install react-native-codegen
```
Comment (cipolleschi) [inline docs/new-architecture-app-intro.md:26]: `yarn add react-native@xxxx` should install the package as well.

File: docs/new-architecture-app-intro.md [hunk 2]
```diff
@@ -54,18 +39,6 @@ cd android && ./gradlew wrapper --gradle-version 7.3 --distribution-type=all
 
 While the AGP version should be updated inside the **top level** `build.gradle` file at the `com.android.tools.build:gradle` dependency line.
 
-If you’re set with it, let’s now install the new Gradle plugin which is distributed through a NPM package called [**`react-native-gradle-plugin`**](https://www.npmjs.com/package/react-native-gradle-plugin). You can do so with:
```
Comment (cipolleschi) [inline docs/new-architecture-app-intro.md:57]: `yarn add react-native@xxxx` should install the package as well.

File: docs/new-architecture-app-modules-android.md [hunk 3]
```diff
@@ -36,10 +36,15 @@ android {
...
                         "PROJECT_BUILD_DIR=$buildDir",
                         "REACT_ANDROID_DIR=$rootDir/../node_modules/react-native/ReactAndroid",
-                        "REACT_ANDROID_BUILD_DIR=$rootDir/../node_modules/react-native/ReactAndroid/build"
+                        "REACT_ANDROID_BUILD_DIR=$rootDir/../node_modules/react-native/ReactAndroid/build",
+                        "NODE_MODULES_DIR=$rootDir/../node_modules"
                 cFlags "-Wall", "-Werror", "-fexceptions", "-frtti", "-DWITH_INSPECTOR=1"
                 cppFlags "-std=c++17"
                 targets "myapplication_appmodules"
+                // Fix for windows limit on number of character in file paths and in command lines
+                if (Os.isFamily(Os.FAMILY_WINDOWS)) {
```
Comment (cipolleschi) [inline docs/new-architecture-app-modules-android.md:45]: I tried to replicate the template here. I think that this is helpful if people develops using windows.

File: docs/new-architecture-appendix.md [hunk 4]
```diff
@@ -27,11 +27,31 @@ You may use the following table as a reference for which types are supported and
 
 You may also find it useful to refer to the JavaScript specifications for the core modules in React Native. These are located inside the `Libraries/` directory in the React Native repository.
 
-## II. Invoking the code-gen during development
+## II. TypeScript to Native Type Mapping
+
+You may use the following table as a reference for which types are supported and what they map to in each platform:
+
+| TypeScript Type                                | Nullable Support?                                        | Android (Java)                       | iOS                                                            | Note                                                                           |
```
Comment (cipolleschi) [inline docs/new-architecture-appendix.md:34]: I'm not sure whether this table is correct or not. Who has a good knowledge of TypeScript to check this?

File: docs/new-architecture-library-android.md [hunk 5]
```diff
@@ -7,41 +7,9 @@ import NewArchitectureWarning from './\_markdown-new-architecture-warning.mdx';
 
 <NewArchitectureWarning/>
 
-Once you have defined the JavaScript specs for your native modules as part of the [prerequisites](new-architecture-library-intro) and followed the Android/Gradle setup, you are now ready to migrate your library to the new architecture. Here are the steps you can follow to accomplish this.
+Once you have defined the JavaScript specs for your native modules as part of the [prerequisites](new-architecture-library-intro), setup the configuration of the Codegen, and followed the Android/Gradle setup, you are now ready to migrate your library to the new architecture. Here are the steps you can follow to accomplish this.
 
-### 1. Configure Codegen in your Gradle File
```
Comment (cipolleschi) [inline docs/new-architecture-library-android.md:12]: Moved to the library intro, now it is shared with iOS

File: docs/new-architecture-app-intro.md (hunk 1 above)
Comment (cortinico) [reply docs/new-architecture-app-intro.md:26]: Yup that's correct 👍 This was old

File: docs/the-new-architecture/_markdown_native_deprecation.mdx [hunk 6]
```diff
@@ -0,0 +1,4 @@
+:::caution
+Native Module and Native Components are legacy technologies bound to the legacy architecture and so, deprecated.
+The New Architecture uses [TurboModule](pillars-turbomodules) and [Fabric Components](pillars-fabric-components) to achieve similar results.
```
[the remark this answers is not in the document]
Comment (cipolleschi) [reply docs/the-new-architecture/_markdown_native_deprecation.mdx:3]: Wasn't the idea to remove those banners with 0.70.0? 🤔

**Comment by cipolleschi** [comment]
@cortinico I reviewed the PR, updated the playbook and added some other fixes.

There are a couple of things left to discuss:
- [ ] Codegen on Android is not working correctly with the new setup
- [ ] What do we want to do with banners and the structure of the website

File: docs/the-new-architecture/_markdown_native_deprecation.mdx (hunk 6 above)
[the remark this answers is not in the document]
Comment (cortinico) [reply docs/the-new-architecture/_markdown_native_deprecation.mdx]: > Wasn't the idea to remove those banners with 0.70.0? 🤔

Nope we should keep those banners as long as we keep on calling the New Architecture experimental. We can reword them as we wish.

**Comment by cortinico** [comment]
> * [ ]  Codegen on Android is not working correctly with the new setup

Addressed in #34298

> * [ ]  What do we want to do with banners and the structure of the website

Let's discuss about this offline? 👍

**Review by arati-1 — COMMENTED** [review]
Hi @cipolleschi I have given minor comments. Take a look.

**Comment by cipolleschi** [comment]
Thanks @arati-1, I applied them to the container PR.

**Pull request merged**